packages/opds-web-client/src/components/__tests__/Collection-test.tsx

```diff
@@ -112,30 +128,37 @@ describe("Collection", () => {
           fulfillBook={fulfillBook}
           indirectFulfillBook={indirectFulfillBook}
           setPreference={setPreference}
-          />,
+        />,
         { context,
           childContextTypes: {
             router: PropTypes.object,
             pathFor: PropTypes.func
           }
         }
       );
-      let viewToggleButtons = wrapper.find(".view-toggle button");
-      expect(viewToggleButtons.length).to.equal(2);
+      const getElements = () => {
+        let viewToggleButtons = wrapper.find(".view-toggle button");
+
+        return [
+          viewToggleButtons,
+          viewToggleButtons.at(0), // gridButton
+          viewToggleButtons.at(1), // listButton
```

Comment:
You could make this an object and then the keys could replace these comments.

Reply:
I initially did that and ended up with `let { viewToggleButtons, gridButton, listButton, books } = getElements();` but in this test when `wrapper.setProps({ preferences });` is called, the components do not update. If you do `wrapper.find(...)` again and refetch the elements, the correct values are there. So I did the object version but couldn't redeclare the variables.

It'd be better if it wasn't an issue to get the element once and get the updated value from it without having to look it up again. Not sure why it's happening for this case.

Reply:
Yeah, that would be better, but seems like it must've been something that changed in enzyme. Anyway, I was just saying you could have this function return an object instead of a list, but you would still need the function.

Reply:
I can't call `let { viewToggleButtons, gridButton, listButton, books } = getElements();` repeatedly though and that's why I made it an array. I'll see if I can figure out something else.

Reply:
Oh interesting, I didn't realize you couldn't do that. Apparently you can put parentheses around it for the subsequent times though - `({ viewToggleButtons, gridButton, listButton, books } = getElements());`

Reply:
It seems like very odd syntax to me but 👍 .